src/main/java/mekanism/common/integration/computer/MethodData.java

```diff
@@ -0,0 +1,95 @@
+package mekanism.common.integration.computer;
+
+import mekanism.common.integration.computer.ComputerMethodFactory.ComputerFunctionCaller;
+import net.minecraftforge.fml.ModList;
+import org.jetbrains.annotations.Nullable;
+
+public record MethodData<T>(String name, MethodRestriction restriction, String[] requiredMods, boolean threadSafe, String[] argumentNames, Class<?>[] argClasses,
+                            Class<?> returnType, Class<?>[] returnExtra, ComputerFunctionCaller<T> handler, @Nullable String methodDescription,
+                            boolean requiresPublicSecurity) {
+
+    public boolean supports(@Nullable T subject) {
+        return restriction.test(subject) && modsLoaded(requiredMods);
+    }
+
+    private boolean modsLoaded(String[] mods) {
+        for (String mod : mods) {
+            if (!ModList.get().isLoaded(mod)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public static <T> Builder<T> builder(String methodName, ComputerFunctionCaller<T> handler) {
+        return new Builder<>(methodName, handler);
+    }
+
+    static String[] NO_STRINGS = new String[0];
+    static Class<?>[] NO_CLASSES = new Class[0];
+
+    public static class Builder<T> {
+
+        private final String methodName;
+        private MethodRestriction restriction = MethodRestriction.NONE;
+        private String[] requiredMods = NO_STRINGS;
+        private boolean threadSafe = false;
+        private String[] argumentNames = NO_STRINGS;
+        private Class<?>[] argClasses = NO_CLASSES;
+        private Class<?> returnType = void.class;
+        private Class<?>[] returnExtra = NO_CLASSES;
+        private final ComputerFunctionCaller<T> handler;
+        private @Nullable String methodDescription = null;
+        private boolean requiresPublicSecurity = false;
+
+        private Builder(String methodName, ComputerFunctionCaller<T> handler) {
+            this.methodName = methodName;
+            this.handler = handler;
+        }
+
+        public MethodData<T> build() {
+            return new MethodData<>(methodName, restriction, requiredMods, threadSafe, argumentNames,argClasses, returnType, returnExtra, handler, methodDescription, requiresPublicSecurity);
+        }
+
+        public Builder<T> restriction(MethodRestriction restriction) {
+            this.restriction = restriction;
+            return this;
+        }
+
+        public Builder<T> requiredMods(String... requiredMods) {
+            this.requiredMods = requiredMods;
+            return this;
+        }
+
+        public Builder<T> threadSafe() {
+            this.threadSafe = true;
+            return this;
+        }
+
+        public Builder<T> arguments(String[] argumentNames, Class<?>[] argClasses) {
+            this.argumentNames = argumentNames;
```

Comment:
Do we want extra validation either here in the builder or maybe in the compact constructor of `MethodData` to validate that we have the same number of argument names as argument classes?